### DISCUSSION TITLE
xwindow title as workspace label

### DISCUSSION THREAD

**Discussion by wackwack** [polybar/polybar#2757] [Q&A]
I'm trying to make my focused bspwm workspace label the xwindow title. My first impulse was to set a key in the xwindow module then access that key in my bspwm module but that passes the token over rather than the token's value. I don't know how tokens are interpreted by polybar so I don't know if that technique could even work.

Is there a way of going about this that I'm totally missing? I haven't had a chance to go through the src code but if I was pointed in the right direction I wouldn't mind doing so.

Maybe a script that makes a request for the focused window the same way that xwindow does?

I just switched to linux last week so I don't fully understand how all the systems interact.

**Comment by patrick96** [comment]
You want the xwindow module to display the name of the current workspace?

For that you could create a bspwm module and set all labels except for `label-focused` to the empty string.

**Reply by wackwack** [reply]
No, I want the bspwm module to display the xwindow title.

I'll try to add an example

**Comment by wackwack** [comment]
![2022-07-02_21-05](https://user-images.githubusercontent.com/104619440/177025727-4cab7b7d-beab-4c66-9491-134da04a1982.png)
On the left you can see my bspwm module and on the right is my xwindow module. 
My xwindow module is displaying Alacrity which is correct as it is what I have open. 
I would like my bspwm `label-focused` to be set to my xwindow title.
For clarity, I currently have `label-empty = -` and `label-occupied =` the checkmark.

Thank you for responding! You have a great project with amazing documentation, hopefully in the future I can contribute

**Reply by patrick96** [reply]
I see what you mean. I don't think that is possible with the built-in modules
